bson/slice_codec.go

```diff
@@ -29,8 +66,9 @@ func (sc *sliceCodec) EncodeValue(ec EncodeContext, vw ValueWriter, val reflect.
 		return vw.WriteNull()
 	}
 
-	// If we have a []byte we want to treat it as a binary instead of as an array.
-	if val.Type().Elem() == tByte {
+	// Treat []byte as binary data, but skip for []int8 since it's a different type
+	// even though byte is an alias for uint8 which has the same underlying type as int8
+	if val.Type().Elem() == tByte && val.Type() != reflect.TypeOf([]int8{}) {
```

Comment:
Can you clarify and add a test illustrating the purpose of `val.Type() != reflect.TypeOf([]int8{}) `? The `bson` test suite passes without this check.
```